### PR TITLE
Improve documentation of struct in regards when to use it

### PR DESCRIPTION
**Pull request by PragTob**
* Based on the explanation given to me in #1130 which helped me
  a lot so I wanted to share it. As this is still my only source
  of information it might as well be that the description given
  here is not 100% correct
* is it just me where all the dates change in all the files on
  one `rake build` ? Or is there a better way to do this/should I not commit those changes?

### CODE REVIEW

File: _gitbook/syntax_and_semantics/structs.md [hunk 1]
```diff
@@ -20,3 +20,5 @@ The differences between a struct and a class are:
 A struct can inherit from other structs and can also includes modules. A struct can be generic, just like a class.
 
 A struct is mostly used for performance reasons to avoid lots of small memory allocations when passing small copies might be more efficient.
+
+So how do you choose between a struct and a class? The rule of thumb is that if no instance variable is ever reassigned, i.e. your type is immutable, you can use a struct, otherwise use a class.
```
Comment (PragTob) [inline _gitbook/syntax_and_semantics/structs.md]: this is the only "real" change

**Comment by PragTob** [comment]
No idea why the build failed :(

**Comment by asterite** [comment]
@PragTob Thanks! I don't know why the build failed either, sometimes travis has issues.

Could you re-send another pull request after pulling some changes I did to the Rakefile? I changed it so that those useless date changes are reverted after the docs are generated, so your new pull request would include just two files (and it could also include just one file, travis is now set up to generate the .html from the .md automatically).

**Comment by PragTob** [comment]
Will totally do that, thanks @asterite !

**Pull request closed without merging**